src/main/java/gregtech/api/recipes/RecipeBuilder.java

```diff
@@ -310,6 +334,109 @@ public R chancedOutputs(List<ChanceEntry> chancedOutputs) {
         return (R) this;
     }
 
+    public R clearChancedOutput() {
+        this.chancedOutputs.clear();
+        return (R) this;
+    }
+
+    /**
+     * Copies the chanced outputs of a Recipe numberOfOperations times, so every chanced output
+     * gets an individual roll, instead of an all or nothing situation
+     *  @param chancedOutputsFrom The original recipe before any parallel multiplication
+     * @param numberOfOperations The number of parallel operations that have been performed
+     */
+
+    public void chancedOutputsMultiply(Recipe chancedOutputsFrom, int numberOfOperations) {
+        for (Recipe.ChanceEntry entry : chancedOutputsFrom.getChancedOutputs()) {
+            int chance = entry.getChance();
+            int boost = entry.getBoostPerTier();
+
+            // Add individual chanced outputs per number of parallel operations performed, to mimic regular recipes.
+            // This is done instead of simply batching the chanced outputs by the number of parallel operations performed
+            IntStream.range(0, numberOfOperations).forEach(value -> {
+                this.chancedOutput(entry.getItemStack(), chance, boost);
+            });
+        }
+    }
+
+    /**
+     *
+     * Appends the passed {@link Recipe} onto the inputs and outputs, multiplied by the amount specified by multiplier
+     * The duration of the multiplied {@link Recipe} is also added to the current duration
+     *
+     * @param recipe The Recipe to be multiplied
+     * @param multiplier Amount to multiply the recipe by
+     * @return the builder holding the multiplied recipe
+     */
+
+    public R append(Recipe recipe, int multiplier) {
+        for (Map.Entry<RecipeProperty<?>, Object> property : recipe.getPropertyValues()) {
+            this.applyProperty(property.getKey().getKey(), property.getValue());
+        }
+
+        // Create holders for the various parts of the new multiplied Recipe
+        List<CountableIngredient> newRecipeInputs = new ArrayList<>();
+        List<FluidStack> newFluidInputs = new ArrayList<>();
+        List<ItemStack> outputItems = new ArrayList<>();
+        List<FluidStack> outputFluids = new ArrayList<>();
+
+        // Populate the various holders of the multiplied Recipe
+        multiplyInputsAndOutputs(newRecipeInputs, newFluidInputs, outputItems, outputFluids, recipe, multiplier);
+
+        // Build the new Recipe with multiplied components
+        this.inputsIngredients(newRecipeInputs);
+        this.fluidInputs(newFluidInputs);
+        this.outputs(outputItems);
+        this.fluidOutputs(outputFluids);
+
+        this.EUt(recipe.getEUt());
+        this.duration(this.duration + recipe.getDuration() * multiplier);
```

Comment:
this method is called by both parallel logics,
in the one used by the multi smelter this.duration is the duration of the previous recipes stored in the builder (64x iron ore + 12 gold ore + 10 diamond ores...)
for the multiplying logic this would,yes, be equal to recipe.getDuration() * multiplier